src/lib/copy.ts

```diff
@@ -0,0 +1,71 @@
+import { Fragment, Slice, Node } from 'prosemirror-model'
+import { TextSelection } from 'prosemirror-state'
+import { EditorView } from 'prosemirror-view'
+import { getTableElementNode, updateSliceWithFullTableContent } from './table'
+import { schema } from '@manuscripts/transform'
+
+export const transformCopied = (slice: Slice, view: EditorView): Slice => {
+    const { state } = view;
+    const { from, to } = state.selection
+    if ((!view.props.handleKeyDown || !state.selection.empty) && slice.content.firstChild?.type === schema.nodes.table) {
+      let tableStart: number | null = null;
+      let tableNode: Node | null = null;
+  
+      // Find the table_element node that contains the copied content
```

Comment:
could this be simplified by using `findParentNodeOfType`?

Reply:
done.